### PR TITLE
improve create and destroy tables error reporting

### PR DESCRIPTION
**Pull request by Charliekenney23**
Improvements:
- Log any creation/deletion errors
- Only attempt to destroy tables that were successfully provisioned

### CODE REVIEW

File: src/dynamodb.js [hunk 1]
```diff
@@ -26,21 +26,41 @@ function buildConfigFromConnection (connection) {
...
+      try {
+        await dynamoClient.createTable(newTable).promise();
+        await dynamoClient.waitFor('tableExists', { TableName }).promise();
+        provisionedTables.push(TableName);
+      } catch (err) {
+        console.error(`Failed to create table "${TableName}"`, err);
+      }
+    })
+  );
+  return provisionedTables;
```
Comment (mdlavin) [inline src/dynamodb.js]: How can the caller tell if there was an error? Should you include the error in the response?

File: src/dynamodb.js [hunk 2]
```diff
@@ -26,21 +26,41 @@ function buildConfigFromConnection (connection) {
...
+    .filter(name => provisionedTables.includes(name));
+
+  await Promise.all(
+    tablesToDestroy
+      .map(async TableName => {
+        try {
+          await dynamoClient.deleteTable({ TableName }).promise();
+          await dynamoClient.waitFor('tableNotExists', { TableName }).promise();
+        } catch (err) {
+          console.error(`Failed to destroy table "${TableName}"`, err);
```
Comment (mdlavin) [inline src/dynamodb.js]: How will you report the failure to destroy table back to the caller?

Comment (Charliekenney23) [reply src/dynamodb.js]: Added a net error to be thrown after all attempts

Comment (Charliekenney23) [reply src/dynamodb.js]: I removed the `provisionedTables` context. It just added mess, `listTables` should filter tables that weren't successfully provisioned anyways.

**Comment by mdlavin** [comment]
I like the new approach to managing failures, this seems like a great improvement. FYI the build failed.

**Comment by Charliekenney23** [comment]
Yeah I saw, fixed 1/2 test failures.

**Pull request merged**